docs/ADRs/0005-python-tooling.md

```diff
@@ -0,0 +1,44 @@
+---
+    title: 0005 Python coding standard and tools
+    adr:
+        author: Berry den Hartog
+        created: 28-Feb-2024
+        status:  accepted
+---
+
+## Context
+
+In modern software development, maintaining code quality is crucial for readability, maintainability, and collaboration.
+Python, being a dynamically typed language, requires robust tooling to ensure code consistency and type safety. Manual
+enforcement of coding standards is time-consuming and error-prone. Hence, adopting automated tooling to streamline this
+process is imperative.
+
+## Decision
+
+We will use these standards and tools for our own projects:
+
+- [google style guide](https://google.github.io/styleguide/pyguide.html)
+- [Black](https://black.readthedocs.io/en/stable/): A code formatter that enforces a consistent code style throughout the
+project.
+- [Pyright](https://microsoft.github.io/pyright/#/): A static type checker for Python, ensuring type safety and reducing
+potential runtime errors.
+- [flake8](https://flake8.pycqa.org/en/latest/): A linting tool that checks for PEP 8 compliance and detects code smells.
+- [isort](https://pycqa.github.io/isort/): an import sorting tool
```

Comment:
I'd like to rather specify the Ruff rules (https://docs.astral.sh/ruff/rules/) that we'd like to apply (instead of Black/flake8/isort):

F -> flake8
I -> Isort
Q -> quotes
UP -> python upgrade

Reply:
btw, the main reason to use Ruff over other tools is that it is incredibly fast (see https://docs.astral.sh/ruff/) and you can even run most of this on safe